### ISSUE TITLE
AggregatedParameter: NotImplementedError: Parameter must be subclassed

### ISSUE THREAD

**Issue by rheinheimer** [pywr/pywr#785]
I am trying to create an aggregated parameter via json that sums the runoff from multiple catchments, following the first json example at https://pywr.github.io/pywr-docs/cookbook/aggregated_parameter.html.

Here is my parameter code:
```
        "Total Runoff": {
            "type": "aggregated",
            "agg_func": "sum",
            "parameters": [
                "node/STN_01 Headflow/Runoff",
                "node/STN_02 Headflow/Runoff",
                "node/STN_03 Headflow/Runoff"
            ]
      }
```
I am trying to access this via a custom Python class as:
`Q_total = model.parameters['Total Runoff'].value(timestep, scenario_index)`

However, I obtain the following error (relevant bits):
```
  File "pywr\parameters\_parameters.pyx", line 46, in pywr.parameters._parameters.Parameter.value
  File "pywr\parameters\_parameters.pyx", line 47, in pywr.parameters._parameters.Parameter.value
NotImplementedError: Parameter must be subclassed
```

Plan B is to pre-process, of course, but any ideas of what is wrong? Note that my component parameters are of type dataframe, reading in from csv files.

**Comment by jetuk** [comment]
Use the `.get_value(scenario_index)` method instead.

**Comment by rheinheimer** [comment]
No, doesn't work. `.get_value(scenario_index)` returns 0.0. I'm thinking maybe it doesn't work for dataframe parameters?

**Comment by rheinheimer** [comment]
Thanks. Low priority, since I can do the aggregation easily in a one-liner: `Q = sum([self.model.parameters[q] for q in my_catchment_parameters])`. I'll report back if/when I get to this again.

**Comment by jetuk** [comment]
The problem is that the parameter system relies on an external entity (the model) to perform the evaluation of all the parameter objects in the correct order. So there's no generic way to perform the entire calculation (including dependent child parameters) from a parameter object itself.
